src/VisualStudio/Core/Impl/SolutionExplorer/DiagnosticItem/CpsDiagnosticItemSource.cs

```diff
@@ -18,27 +20,24 @@ internal sealed partial class CpsDiagnosticItemSource : BaseDiagnosticAndGenerat
     private readonly IVsHierarchyItem _item;
     private readonly string _projectDirectoryPath;
 
-    /// <summary>
-    /// The analyzer reference that has been found. Once it's been assigned a non-null value, it'll never be assigned null again.
-    /// </summary>
-    private AnalyzerReference? _analyzerReference;
```

Comment:
Moved to base class. note that this comment is important. it was an invariant of this subclass that it would only set this once. We check for thsi invariant in the base class now.

Reply:
You should be able to use `[DisallowNull]` for this.